astropy/wcs/wcs.py

```diff
@@ -913,10 +913,11 @@ def _read_det2im_kw(self, header, fobj, err=0.0):
                 dis = header[distortion].lower()
                 if dis == "lookup":
                     del header[distortion]
-                    assert isinstance(fobj, fits.HDUList), (
-                        "An astropy.io.fits.HDUList"
-                        "is required for Lookup table distortion."
-                    )
+                    if not isinstance(fobj, fits.HDUList):
+                        raise TypeError(
```

Comment:
This is an API change. Maybe OK since it is unusual for calling code to catch an `AssertionError`, but still.

Reply:
Change to `AssertionError`.